casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -66,9 +66,10 @@ object InterpreterUtil {
       internalMap: Map[BlockHash, BlockMessage],
       emptyStateHash: StateHash,
       knownStateHashes: Set[StateHash],
-      computeState: (StateHash, Seq[Deploy]) => Either[DeployError, Checkpoint])
-    : (Checkpoint, Set[StateHash]) = {
-    val (postStateHash, updatedStateHashes) =
+      computeState: (StateHash,
+                     Seq[Deploy]) => Either[DeployError, (Checkpoint, Vector[DeployCost])])
+    : (Checkpoint, Set[StateHash], Vector[DeployCost]) = {
+    val (postStateHash, updatedStateHashes, postStateCost) = //TODO: what to do with the postStateCost -- ignore?
```

Comment:
This is a good question. Ideally it does not actually take any work to deduce the parent post state because the parents need to be independent, so you should be able to just "merge" the tuplepaces. Of course we don't have any way of doing that right now, so we just repeat the calculations on top of each other, which is potentially an intensive task. Ultimately though we do not want it to cost extra to produce multiparent blocks because they are good for the network since they enable a fork-join parallelism model in the blockchain itself (multiparent blocks are a join bringing to previously concurrent forks together).

For now the easiest thing to do is to probably just ignore this `postStateCost` and we should have a TODO item to revisit the way we merge parent states.